utils/src/main/kotlin/Utils.kt

```diff
@@ -392,6 +392,18 @@ fun Throwable.collectMessages(): List<String> {
     return messages
 }
 
+/**
+ * Recursively collect the messages of this [Throwable] and all its causes and join them to a single [String].
+ */
+fun Throwable.collectMessagesAsString() =
+        collectMessages().let { messages ->
+            if (messages.size == 1) {
+                messages.first()
+            } else {
+                messages.joinToString(", caused by ") { "\"$it\"" }
```

Comment:
Another nit: Should we still align to how `printStackTrace()` writes "Caused by:" in a new line, see r.g. https://ci.appveyor.com/project/heremaps/oss-review-toolkit/build/1.0.3415#L4375?

Reply:
Good idea, done.